### PR TITLE
Capture GC times in performance tests

### PR DESCRIPTION
**Pull request by wolfs**
By updating Gradle profiler to `0.15.0-alpha10`, we can capture GC times in the Gradle profiler reports for our performance tests.

### CODE REVIEW

**Comment by wolfs** [comment]
We now set the teamcity status to include the test name if only one test did execute:
image

https://builds.gradle.org/buildConfiguration/Gradle_Util_Performance_AdHocPerformanceScenarioLinux/38626748

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/GradleProfilerReporter.java [hunk 1]
```diff
@@ -17,57 +17,96 @@
...
-        }
+    public BenchmarkResultCollector getResultCollector() {
         return resultCollector;
     }
 
     @Override
     public void close() {
     }
+
+    private static class DelegatingReportGenerator extends AbstractGenerator {
```
Comment (lptr) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/GradleProfilerReporter.java:81]: Shouldn't this be called a `CompositeGenerator`?

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/PerformanceScenario.java [hunk 2]
```diff
@@ -36,6 +36,10 @@ public String getClassName() {
         return className;
     }
 
+    public String getSimpleClassName() {
+        return className.substring(className.lastIndexOf('.') + 1);
+    }
```
Comment (lptr) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/PerformanceScenario.java:41]: Let's see how this works out, but I think we should just make this part of the display name, and use it everywhere.

**Review by lptr — APPROVED** [review]
LGTM, left a couple comments.

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/GradleProfilerReporter.java (hunk 1 above)
Comment (wolfs) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/GradleProfilerReporter.java]: Renamed.

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/PerformanceScenario.java (hunk 2 above)
Comment (wolfs) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/PerformanceScenario.java:41]: I added it now to the display name. I'll run a regression performance test suite, to see what the full report looks like. In my small test it doesn't look to bad: 
image

**Pull request merged**